ideas/279-snt-voting-dapp.md

```diff
@@ -0,0 +1,106 @@
+---
+id: 279-snt-voting-dapp
+title: SNT Voting Dapp
+status: draft
+created: 2018-06-05
+category: dapp
+contributors:
+    - Barry 
+    - 3esmit
+    - rramos
+exit-criteria: yes
+success-metrics: yes
+clear-roles: yes
+future-iteration: yes
+roles-needed:
+    - QA
+    - UXR
+    - Designer
+okrs:
+    -
+    -
+---
+
+## Preamble
+
+    Idea: 279
+    Title: SNT Voting Dapp
+    Status: Draft
+    Created: 2018-05-21
+    Requires (*optional): 172-topic-democracy
+
+## Summary
+
+Provide Status users with a dApp to vote on ideas using their SNT without affecting their token balance.
+
+## Swarm Participants
+
+- Lead Contributor: @3esmit
+- Contributor: @Barry
+- Contributor: @rramos
+- Contributor: @3esmit
+- QA:
+- PM (required for user-facing): @rachelhamlin
+- UX(R) (required for user-facing swarms):
+
+## Product Overview
+
+Status.im has ideas for a DApp idea collective commons, and we want the SNT Holders to have the ability to vote on these and decide a winner
```

Comment:
This isn't super clear to me, maybe we can rewrite this paragraph (Dapp idea collective commons, winner)?

Also, can we enumerate what type of voting capabilities we have or plan? Examples:
- 1 SNT address 1 vote
- 1 identity (?) 1 vote
- N SNT address N votes
- N SNT address sqrt(N) votes (quadric voting)

Reply:
This is a good question. I have been in favor of the SNT allocation method. Where an address can choose how much of its SNT to allocate to each item they vote on. This way the vote expresses not just the features SNT holders want built but the levels of interest in each which is valuable in determining priority of development.

An example of this has already been implemented in the Aragon survey so there is a good amount of learnings from them we can leverage here. ![Image of Aragon Survey](https://cdn-images-1.medium.com/max/2000/1*o7yRBPfaXV7lGQRbQ5eXmQ.jpeg) https://survey.aragon.org

Would love to hear thoughts on this. @rachelhamlin @richard-ramos

Reply:
Sounds like the DApp collective commons bit is from the sPrize idea, where people would vote on the use cases we want to launch with that initiative. 

Might still be used there, but this is also for Status features and organizational decisions, no?

Looking at this example on Aragon, I'm inclined to support allocation too. It looks like a really human-friendly way to let people prioritize. 

<img src="https://user-images.githubusercontent.com/15470435/41424117-1274827a-6ff5-11e8-9970-b5bbe9b55c73.png" width=300>

Reply:
Indeed, allocation method seems to be the most user friendly option, I think that with this method we should be able to visually support different types of voting: 
- Simple voting with Approve/Reject: UI would show two choices, and the code logic would allocate 100% of the tokens to one option
- Multiple choice (only one option can be selected): same as previous option, but instead of having two choices, we'd have N.
- Token Allocation, like Aragon's, which would be the base behaviour.
This would make voting on features and decisions more complete as we could handle different use cases.